flask_security/core.py

```diff
@@ -425,6 +425,9 @@ def init_app(self, app, datastore=None, register_blueprint=True,
         """
         datastore = datastore or self.datastore
 
+        email_sender = app.config.get('MAIL_DEFAULT_SENDER')
+        _default_config['EMAIL_SENDER'] = email_sender or 'no-reply@localhost'
```

Comment:
We should not modify the `_default_config`, but instead directly `setdefault` on `app.config` with new value.

```python
# add after _default_config
_default_config_from_app = {
    'MAIL_DEFAULT_SENDER': 'EMAIL_SENDER',
}

for app_key, key in _default_config_from_app.items():
    value = app.config.get(app_key)
    if value:
        app.config.setdefault('SECURITY_' + key, value)
```